### ISSUE TITLE
How to debug Binaryen with Windows and Visual Studio ?

### ISSUE THREAD

**Issue by haley-dot-yang** [WebAssembly/binaryen#6398]
Hi, I try to debug Binaryen tool wasm-split on windows with Visual Studio 2022. The CMD I used as follows:
```shell
git clone https://github.com/WebAssembly/binaryen.git
git submodule init
git submodule update
mkdir build
cd build
cmake ..
cmake --build . --config Debug
```
While compiling in CMD, there is an error log:
```shell
binaryen\src\parser\wat-parser.cpp : fatal  error C1128: number of sections exceeded object file format
limit: compile with /bigobj [D:\GithubRepos\binaryen\build\src\parser\parser.vcxproj]
cl : command line  error D8040: error creating or communicating with child process [D:\GithubRepos\binaryen\build\src\p
arser\parser.vcxproj]
```
If open binaryen.sln in VS2022, choose wasm-split target to build in x64 Debug mode, the same error
![image](https://github.com/WebAssembly/binaryen/assets/144108324/9c661a7e-2e84-497c-807f-2589535da13e)
then I add /bigobj in Additional Options
![image](https://github.com/WebAssembly/binaryen/assets/144108324/305bc836-dca6-4f35-914d-e14687a954a5)
But the error are still unresolved. However, when I switch to Release mode, it can successfully build.
Can I get any suggestions here? Thank you.

**Comment by kripken** [comment]
```
binaryen\src\parser\wat-parser.cpp : fatal  error C1128: number of sections exceeded object file format
```
Hmm, I'm not familiar with Visual Studio but I'm guessing that means we are running into some internal limitation there.

I believe this file also compiles very slowly atm due to the many templates - is that right @tlively ? I seem to recall you might have had an idea for how to improve that. If so that plan might also help avoid this VS limitation I'd hope.

**Comment by tlively** [comment]
Yeah, splitting things up to improve the compile time might also fix this problem.

In the meantime, it looks like there is a [`/bigobj`](https://learn.microsoft.com/en-us/cpp/build/reference/bigobj-increase-number-of-sections-in-dot-obj-file?view=msvc-170) flag that should resolve this.

**Comment by haley-dot-yang** [comment]
As I posted in the last figure, I added the compiling flag `/bigobj` in Visual Studio, but it does not help to resolve this problem. 
Now I switched to VS Code, and it can debug now.  Here is my VSCode CMake Project Status Settings
![image](https://github.com/WebAssembly/binaryen/assets/144108324/c79186e8-03de-4eca-b0c4-7fd2b69eb739)
It confused me so much: when I run Debug in Visual Studio, it failed; however, if I debug in VS Code with Visual Studio Toolchain (as shown in the figure in this comment), it works. Orz
